### PR TITLE
Add `generate_openapi_spec` method to `Module`.

### PR DESCRIPTION
**Pull request by rohinb2**
---

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## Description for Pull Request
> 
> ### TL;DR
> Added a method to generate OpenAPI spec for a module, updated dependencies, and added tests for OpenAPI spec generation.
> 
> ### What changed
> - Added a `generate_openapi_spec` method to the `Module` class in `module.py` to generate an OpenAPI spec for the module.
> - Updated dependencies in `setup.py` to include `apispec` and in `tests/requirements.txt` to include `openapi-core`.
> - Added tests in `test_module.py` to test the generation of OpenAPI spec for a module.
> 
> ### How to test
> 1. Ensure the `generate_openapi_spec` method in `module.py` generates the OpenAPI spec correctly for the module.
> 2. Check that the updated dependencies in `setup.py` and `tests/requirements.txt` are installed successfully.
> 3. Run the tests in `test_module.py` to verify the OpenAPI spec generation for a module.
> 
> ### Why make this change
> - The addition of the `generate_openapi_spec` method allows for easy generation of OpenAPI specs for modules.
> - Updating dependencies ensures that the required packages for OpenAPI spec generation are available.
> - Adding tests ensures that the OpenAPI spec generation functionality is tested and working as expected.
</details>

### CODE REVIEW

**Comment by rohinb2** [comment]
* **#568** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/568?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/module.py [hunk 1]
```diff
@@ -942,6 +945,94 @@ def _extract_pointers(raw_cls_or_fn: Union[Type, Callable], reqs: List[str]):
...
+        """
+
+        spec_name = spec_name or self.name
+
+        if not spec_name:
+            raise ValueError(
+                "Module must have a name to generate an OpenAPI spec. Try saving the module first."
+            )
+
+        cls = self.__class__
```
Comment (dongreenberg) [inline runhouse/resources/module.py:969]: I think it's a bad practice to name a variable `cls`, sort like naming a random variable `self`

**Review by dongreenberg — APPROVED** [review]
I think this look good, but whether the spec is structured correctly/usefully is going to depend on how it looks when we try to use it to stand up a swagger UI for testing.

File: runhouse/resources/module.py (hunk 1 above)
Comment (rohinb2) [reply runhouse/resources/module.py:969]: resolved, was a remnant from an earlier version where I was reusing that, killed the variable off now

**Comment by rohinb2** [comment]
> I think this look good, but whether the spec is structured correctly/usefully is going to depend on how it looks when we try to use it to stand up a swagger UI for testing.
> 
> 

I think this is standard OpenAPI spec actually so it should be good if @mkandler uses some standard OpenAPI display for it

**Pull request merged**